llvm/include/llvm/IR/InstrTypes.h

```diff
@@ -1058,6 +1058,17 @@ class CmpInst : public Instruction {
   static CmpInst *Create(OtherOps Op, Predicate predicate, Value *S1,
                          Value *S2, const Twine &Name, BasicBlock *InsertAtEnd);
 
+  /// Construct a compare instruction, given the opcode, the predicate,
+  /// the two operands and the instruction to copy the flags from. Optionally
+  /// (if InstBefore is specified) insert the instruction into a BasicBlock
+  /// right before the specified instruction. The specified Instruction is
+  /// allowed to be a dereferenced end iterator. Create a CmpInst
+  static CmpInst *CreateWithCopiedFlags(OtherOps Op, Predicate Pred, Value *S1,
```

Comment:
```suggestion
  /// allowed to be a dereferenced end iterator.
  /// Create a CmpInst
  static CmpInst *CreateWithCopiedFlags(OtherOps Op, Predicate Pred, Value *S1,
```

Reply:
Will include this suggestion in a follow-up pr to do some cleanup.

Reply:
See #86441